pulsar-client/src/main/java/org/apache/pulsar/client/impl/BatchMessageContainerImpl.java

```diff
@@ -197,6 +197,26 @@ public boolean isMultiBatches() {
 
     @Override
     public OpSendMsg createOpSendMsg() throws IOException {
+        if (messages.size() == 1) {
+            MessageImpl<?> msg = messages.get(0);
+            messageMetadata = msg.getMessageBuilder();
+            ByteBuf encryptedPayload = producer.encryptMessage(
+                messageMetadata,
+                producer.applyCompression(msg.getDataBuffer()));
+            if (encryptedPayload.readableBytes() > ClientCnx.getMaxMessageSize()) {
+                discard(new PulsarClientException.InvalidMessageException(
+                    "Message size is bigger than " + ClientCnx.getMaxMessageSize() + " bytes"));
+                return null;
+            }
+            ByteBufPair cmd = producer.sendMessage(producer.producerId, messageMetadata.getSequenceId(),
+                1, messageMetadata, encryptedPayload);
+            final OpSendMsg op;
+            op = OpSendMsg.create(msg, cmd, messageMetadata.getSequenceId(), firstCallback);
+            op.setNumMessagesInBatch(1);
```

Comment:
I don't think we have to set this header

Reply:
Same as `OpSendMsg#batchSizeByte` below.  But the because the `numMessagesInBatch` default value  is `1`, it seems that it doesn't matter if we set it or not.